samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -0,0 +1,205 @@
+#include "opencv2/core/utility.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/highgui.hpp"
+#include <iostream>
+
+using namespace std;
+using namespace cv;
+
+string getGstDemuxPlugin(CommandLineParser &parser);
+string getGstAvDecodePlugin    (CommandLineParser &parser);
+void checkSupportArguments(CommandLineParser &parser);
+
+int main(int argc, char *argv[])
+{
+    const string keys =
+          "{h help usage ? |           | print help message   }"
+          "{s show         |false      | show frames }"
+          "{p pipeline     |gst-default| pipeline name  (supported: 'gst-default', 'gst-vaapi', 'gst-libav', 'ffmpeg') }"
+          "{ct container   |mp4        | container name (supported: 'mp4', 'mov', 'avi', 'mkv') }"
+          "{cd codec       |h264       | codec name     (supported: 'h264', 'h265', 'mpeg4', 'mjpeg', 'vp8') }"
+          "{f file path    |           | path to file }";
+
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("This program shows how to read a video file with GStreamer pipeline with OpenCV.");
+
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    if (!parser.check())
+    {
+        parser.printErrors();
+        return 0;
+    }
+
+    bool   arg_show;      // show frames on the screen
+    string arg_pipeline,  // GStreamer pipeline type
+           arg_container, // container type
+           arg_codec,     // codec type
+           arg_file_name, // path to videofile
+           codec_name,
+           meta_info;
+    VideoCapture cap;
+
+    arg_show      = parser.get<bool>  ("show");
```

Comment:
If you combine variable initialization and declaration you will save 5 lines of code and make this part of program less error-prone:
```.cpp
bool arg_show = parser.has("show");
```